examples/vision/image_captioning.py

```diff
@@ -409,38 +409,65 @@ def _compute_loss_and_acc(self, batch_data, training=True):
                 # 5. Pass the encoder outputs, sequence inputs along with
                 # mask to the decoder
                 batch_seq_pred = self.decoder(
-                    batch_seq_inp, encoder_out, training=training, mask=mask
+                    batch_seq_inp, encoder_out, training=True, mask=mask
                 )
 
-                # 6. Calculate loss and accuracy
-                loss = self.calculate_loss(batch_seq_true, batch_seq_pred, mask)
-                acc = self.calculate_accuracy(batch_seq_true, batch_seq_pred, mask)
-
-                # 7. Update the batch loss and batch accuracy
-                batch_loss += loss
-                batch_acc += acc
+                # 6. Update loss and accuracy
+                loss += self.calculate_loss(batch_seq_true, batch_seq_pred, mask)
+                acc += self.calculate_accuracy(batch_seq_true, batch_seq_pred, mask)
 
-            # 8. Get the list of all the trainable weights
+            # 7. Get the list of all the trainable weights
             train_vars = (
                 self.encoder.trainable_variables + self.decoder.trainable_variables
             )
 
-            # 9. Get the gradients
+            # 8. Get the gradients
             grads = tape.gradient(loss, train_vars)
```

Comment:
The update here is for each caption. Either keep the `batch_loss` where you accumulate the loss for each caption or just use another name for per caption loss

Reply:
I think using different variable names would be more appropriate. I've changed the variable names to clearly distinguish between the losses.